src/System.Management.Automation/CoreCLR/CorePsExtensions.cs

```diff
@@ -1044,6 +1044,11 @@ private static string InternalGetFolderPath(SpecialFolder folder)
             string folderPath = null;
 
 #if UNIX
+            string envHome = System.Environment.GetEnvironmentVariable("HOME");
```

Comment:
Looking at the code where "USERPROFILE" env var is used, it appears to handle the case where it's null already